core-bundle/src/Resources/contao/controllers/BackendPassword.php

```diff
@@ -81,21 +81,46 @@ public function run()
 			// Save the data
 			else
 			{
+				$this->loadDataContainer('tl_user');
+
+				$dc = new \DC_Table('tl_user');
+				$dc->id = $this->User->id;
+
```

Comment:
I would prefer to keep this code in the `elseif` and `else` condition, even though this means duplicating it. Loading a DCA and instantiating a new DC object are expensive operations that are not necessary for the `if` condition.